server/src/main/java/org/elasticsearch/common/blobstore/support/FilterBlobContainer.java

```diff
@@ -79,6 +79,11 @@ public void writeBlobAtomic(String blobName, BytesReference bytes, boolean failI
         delegate.writeBlobAtomic(blobName, bytes, failIfAlreadyExists);
     }
 
+    @Override
+    public void writeBlob(String blobName, BytesReference bytes, boolean failIfAlreadyExists) throws IOException {
```

Comment:
if I'm reading correctly the interface definition:

```
    default void writeBlob(String blobName, BytesReference bytes, boolean failIfAlreadyExists) throws IOException {
        writeBlob(blobName, bytes.streamInput(), bytes.length(), failIfAlreadyExists);
    }
```

it calls another `writeBlob()` which is overriden by the FilterBlobContainer. So maybe this PR is unnecessary?

Reply:
If an `FsBlobContainer` is wrapped into a `FilterBlobContainer`, you can expect that executing the method `writeBlob(String, BytesReference, boolean)` on that filtered blob container would delegate to the `writeBlob(String, BytesReference, boolean)` method of the `FsBlobContainer`, but it's not what will happen. Instead the method `writeBlob(String, InputStream, long, boolean)` will be executed on the FS implementation, which provides an implemention for both writeBlob methods.

Reply:
@tlrx I am still confused about what you said. But interestingly this [PR](https://github.com/elastic/elasticsearch/pull/100177/files#diff-dae4817aa9bc4a52cd6355b3382f597877d89d989f869ebea8e4006be2115d45L143) removed the `writeBlob(String, BytesReference, boolean)` from the base interface yesterday, so maybe this PR is unnecessary now due to that.

Reply:
Sorry if I'm not clear 😓 

I debugged a test where an instance of `FsBlobContainer` was wrapped into a `FilterBlobContainer`, something I often do in tests to assert or to simulate a specific behavior of the `BlobContainer`.

The code I was debugging was supposed to call the method `writeBlob(OperationPurpose, String, BytesReference, boolean)` on the `FilterBlobContainer` instance at some point of the execution. But the method was never called, because  `FilterBlobContainer` by default delegates to the other write blob method, not the one I was expected. I wasted a bit of time on this I found this trappy, so wanted to fix it in this PR before I forget.

Here some test that illustrates the trap I felt into:
```java
public void testMe() throws IOException {
        var blobContainer = mock(BlobContainer.class);
        var wrapper = new FilterBlobContainer(blobContainer) {
            @Override
            protected BlobContainer wrapChild(BlobContainer child) {
                throw new AssertionError("not tested here");
            }
        };
        wrapper.writeBlob(OperationPurpose.SNAPSHOT, "one", new ByteArrayInputStream(randomByteArrayOfLength(12)), 12L, randomBoolean());
        wrapper.writeBlob(OperationPurpose.SNAPSHOT, "two", new BytesArray(randomByteArrayOfLength(12)), randomBoolean());
        verify(blobContainer, times(1)).writeBlob(any(), anyString(), any(InputStream.class), anyLong(), anyBoolean()); // fails with times == 2
        verify(blobContainer, times(1)).writeBlob(any(), anyString(), any(BytesReference.class), anyBoolean());
    }
```